scio-core/src/main/scala/com/spotify/scio/ScioContext.scala

```diff
@@ -454,6 +455,87 @@ class ScioContext private[scio] (
     o.setScioVersion(BuildInfo.version)
   }
 
+  {
+    import org.apache.hadoop.conf.Configuration
+    import com.google.cloud.hadoop.fs.gcs.{GoogleHadoopFileSystemConfiguration => GfsConfig}
+    import com.google.cloud.hadoop.gcsio.GoogleCloudStorageReadOptions
+
+    try {
+      // If Hadoop is on the classpath, try to parse default gcs-connector options
+      val config = new Configuration()
+      val o = optionsAs[GcsOptions]
+
+      o.setGoogleCloudStorageReadOptions(
```

Comment:
There's a method to parse all this for us but it's not public: https://github.com/GoogleCloudDataproc/hadoop-connectors/blob/v3.0.4/gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemConfiguration.java#L645-L660

I do have a PR up to make it public so hopefully it can be dropped in in a future release.

Reply:
Link the PR in the code somewhere so we can find it later